File: Pie02/Pie02IronPython/Rules/ImportRules.py
from Irony.Parsing import *

class ImportRules(object):
    
    def __init__(self, grammar):
        self.ImportedName = NonTerminal("qualified_import")
        self.ImportedName.Rule = grammar.QualifiedIdentifier
        self.ImportList = NonTerminal("import_list")
        self.ImportList.Rule = grammar.MakePlusRule(self.ImportList, grammar.Comma, self.ImportedName)
        self.ImportDeclaration = NonTerminal("import_declaration")
        self.ImportDeclaration.Rule = grammar.Import + self.ImportList + grammar.Eos
        

File: Pie02/Pie02CS/Rules/ImportRules.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Irony.Parsing;

namespace Pie02CS.Rules
{
    public class ImportRules
    {
        public NonTerminal ImportedName;
        public NonTerminal ImportList;
        public NonTerminal ImportDeclaration;

        public ImportRules(PieGrammar grammar)
        {
            ImportedName = new NonTerminal("qualified_import");
            ImportedName.Rule = grammar.QualifiedIdentifier;

            ImportList = new NonTerminal("import_list");
            ImportList.Rule = grammar.MakePlusRule(ImportList, grammar.Comma, ImportedName);

            ImportDeclaration = new NonTerminal("import_declaration");
            ImportDeclaration.Rule = grammar.ToTerm("import") + ImportList + grammar.Eos;
        }
    }
}
